Source/Core/Core/NetworkCaptureLogger.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Core/NetworkCaptureLogger.h"
+
+#include "Common/File.h"
+#include "Common/FileUtil.h"
+#include "Core/Config/MainSettings.h"
+#include "Core/ConfigManager.h"
+
+namespace Core
+{
+NetworkCaptureLogger::NetworkCaptureLogger() = default;
+NetworkCaptureLogger::~NetworkCaptureLogger() = default;
+
+void DummyNetworkCaptureLogger::LogRead(const void* data, std::size_t length)
+{
+}
+
+void DummyNetworkCaptureLogger::LogWrite(const void* data, std::size_t length)
+{
+}
+
+NetworkCaptureType DummyNetworkCaptureLogger::GetCaptureType() const
+{
+  return NetworkCaptureType::None;
+}
+
+void BinarySSLCaptureLogger::LogRead(const void* data, std::size_t length)
+{
+  if (!Config::Get(Config::MAIN_NETWORK_SSL_DUMP_READ))
```

Comment:
Is this check actually useful, considering you're already checking those config settings in the logger instance getter?